src/util/pointer.h

```diff
@@ -257,3 +257,12 @@ class SharedBuffer
 	unsigned int *refcount;
 };
 
+// This class is not thread-safe!
+class IntrusiveReferenceCounted {
+public:
+	virtual ~IntrusiveReferenceCounted() = default;
+	void grab() noexcept { ++m_refcount; }
+	void drop() noexcept { if (--m_refcount == 0) delete this; }
+private:
+	u32 m_refcount = 1;
+};
```

Comment:
Add `DISABLE_CLASS_COPY(IntrusiveReferenceCounted)`. Or implement suitable copy and move constructors and assignment operators.

Reply:
Thanks, I implemented custom copying and assignment.

Reply:
(For rule of 5, move constr and assign would be nice too. Semantics are probably the same as on copy.)

Reply:
Won't the no-op copy construction/assignment implementations be used when moving construction/assignment is needed?

Reply:
Oh, yes, you're right.